blockchain/blockchain_test.go

```diff
@@ -1644,3 +1658,403 @@ func TestBlockchain_WriteFullBlock(t *testing.T) {
+
+	header.Timestamp, err = common.ParseUint64orHex(&timestamp)
+	if err != nil {
+		return nil, err
+	}
+
+	header.ExtraData = types.StringToBytes(dat["extraData"].(string)) //nolint:forcetypeassert
+
+	nonce := dat["nonce"].(string) //nolint:forcetypeassert
+
+	nonceNumber, err := common.ParseUint64orHex(&nonce)
+	if err != nil {
+		return nil, err
+	}
+
+	header.Nonce = types.Nonce{byte(nonceNumber)}
+
+	header.Hash = types.StringToHash(dat["hash"].(string)) //nolint:forcetypeassert
+
+	for _, transactionJSON := range dat["transactions"].([]interface{}) { //nolint:forcetypeassert
+		tr := transactionJSON.(map[string]interface{}) //nolint:forcetypeassert
+		transaction := &types.Transaction{}
+		transaction.Hash = types.StringToHash(tr["hash"].(string)) //nolint:forcetypeassert
+		nonce := tr["nonce"].(string)                              //nolint:forcetypeassert
+
+		nonceNumber, err := common.ParseUint64orHex(&nonce)
+		if err != nil {
+			return nil, err
+		}
+
+		transaction.Nonce = nonceNumber
+
+		transaction.From = types.StringToAddress(tr["from"].(string)) //nolint:forcetypeassert
+		addr := types.StringToAddress(tr["to"].(string))              //nolint:forcetypeassert
+		transaction.To = &addr
+
+		value := tr["value"].(string) //nolint:forcetypeassert
+
+		valueNumber, err := common.ParseUint256orHex(&value)
+		if err != nil {
+			return nil, err
+		}
+
+		transaction.Value = valueNumber
+
+		gasPrice := tr["gasPrice"].(string) //nolint:forcetypeassert
+
+		gasPriceNumber, err := common.ParseUint256orHex(&gasPrice)
+		if err != nil {
+			return nil, err
+		}
+
+		transaction.GasPrice = gasPriceNumber
+
+		transaction.Input = []byte(tr["input"].(string)) //nolint:forcetypeassert
+
+		v := tr["v"].(string) //nolint:forcetypeassert
+
+		vNumber, err := common.ParseUint256orHex(&v)
+		if err != nil {
+			return nil, err
+		}
+
+		transaction.V = vNumber
+
+		r := tr["r"].(string) //nolint:forcetypeassert
+
+		rNumber, err := common.ParseUint256orHex(&r)
+		if err != nil {
+			return nil, err
+		}
+
+		transaction.R = rNumber
+
+		s := tr["s"].(string) //nolint:forcetypeassert
+
+		sNumber, err := common.ParseUint256orHex(&s)
+		if err != nil {
+			return nil, err
+		}
+
+		transaction.S = sNumber
+
+		chainID := tr["chainId"].(string) //nolint:forcetypeassert
+
+		chainIDNumber, err := common.ParseUint256orHex(&chainID)
+		if err != nil {
+			return nil, err
+		}
+
+		transaction.ChainID = chainIDNumber
+
+		txType := tr["type"].(string) //nolint:forcetypeassert
+
+		txTypeNumber, err := common.ParseUint64orHex(&txType)
+		if err != nil {
+			return nil, err
+		}
+
+		transaction.Type = types.TxType(txTypeNumber)
+
+		transactions = append(transactions, transaction)
+	}
+
+	return &types.FullBlock{Block: &types.Block{Header: header, Transactions: transactions}}, nil
+}
+
+func customJSONReceiptsUnmarshall(tb testing.TB, jsonData []byte) ([]*types.Receipt, error) {
+	tb.Helper()
```

Comment:
I'd consider removing `tb` parameter:

```suggestion
func customJSONReceiptsUnmarshall(jsonData []byte) ([]*types.Receipt, error) {
```